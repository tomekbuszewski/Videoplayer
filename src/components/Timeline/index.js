/**
 * @flow
 * @author tomekbuszewski
 * @since 2018-11-26
 */

import * as React from 'react';
import inject from 'react-jss';
import c from 'classnames';

import Highlight from '@src/containers/Highlight';

import getPercentage from '@services/getPercentage';
import secondsToTime from '@services/secondsToTime';

import styles from './index.styles';

type Props = {
  classes: Object,
  duration: number,
  highlights: Object[] | null,
  position: number,
  onClick: Function,
}

const Timeline = (props: Props) => {
  const {
    classes,
    duration,
    highlights,
    position,
    onClick,
  } = props;

  return (
    <div className={classes.Timeline}>
      <div className={classes.Track} />
      <input
        className={c(classes.Track, classes.Range)}
        type="range"
        min={0}
        max={duration}
        defaultValue={position}
        onChange={(e) => { onClick(e.target.value); }}
        step={1}
      />
      <div className={classes.Pointer} style={{ left: getPercentage(duration, position, true) }}>
        <span className={classes.Timer}>{secondsToTime(position)}</span>
      </div>
      {highlights && highlights.map((item: Object) => (
        <Highlight
          key={`highlight-${item.time}-${duration}`}
          className={classes.Pointer}
          style={{ left: getPercentage(duration, item.time, true) }}
          onClick={() => { onClick(item.time); }}
          image={item.image}
          caption={item.caption}
        />
      ))}
    </div>
  );
};

export default inject(styles)(Timeline);
