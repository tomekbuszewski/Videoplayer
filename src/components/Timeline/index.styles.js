import { COLOR_WHITE } from '@config/colors';
import {
  OPACITY_MEDIUM,
  STEP_EXTRASMALL,
  STEP_SMALL,
  GET_STEP, STEP_MEDIUM,
} from '@config/visuals';

export default {
  Timeline: {
    width: '100%',
    position: 'relative',
  },

  Track: {
    position: 'absolute',
    width: '100%',
    top: 0,
    left: 0,
    transform: 'translateY(-50%)',
    height: GET_STEP(STEP_EXTRASMALL),
    background: COLOR_WHITE,
    borderRadius: GET_STEP(STEP_EXTRASMALL),
    opacity: OPACITY_MEDIUM,
  },

  Range: {
    opacity: 0,
    width: `calc(100% + ${GET_STEP(STEP_SMALL)}px)`,
    left: -GET_STEP(STEP_SMALL) / 2,
  },

  Timer: {
    color: COLOR_WHITE,
    position: 'absolute',
    top: GET_STEP(STEP_MEDIUM),
  },

  Pointer: {
    top: '50%',
    left: 0,
    transform: 'translate(-50%, -50%)',
    position: 'absolute',
    background: COLOR_WHITE,
    borderRadius: GET_STEP(STEP_SMALL),
    pointerEvents: 'none',

    width: GET_STEP(STEP_SMALL),
    height: GET_STEP(STEP_SMALL),
  },
};
