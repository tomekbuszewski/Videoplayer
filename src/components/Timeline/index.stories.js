import * as React from 'react';

import { withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import Timeline from '.';

storiesOf('Atoms', module)
  .addDecorator(withKnobs)
  .add(
    'Timeline',
    (() => <Timeline />),
  );
