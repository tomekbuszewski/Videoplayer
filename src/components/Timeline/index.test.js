import * as React from 'react';
import renderer from 'react-test-renderer';

import Timeline from './';

describe('Timeline tests', () => {
  it ('should match snapshot', () => {
    const component = renderer.create(<Timeline />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
