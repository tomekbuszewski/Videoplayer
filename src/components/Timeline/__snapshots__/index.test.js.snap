// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline tests should match snapshot with highlights 1`] = `
<div
  className="Timeline-Timeline-0-1-1"
>
  <div
    className="Timeline-Track-0-1-2"
  />
  <input
    className="Timeline-Track-0-1-2 Timeline-Range-0-1-3"
    defaultValue={0}
    max={100}
    min={0}
    onChange={[Function]}
    step={1}
    type="range"
  />
  <div
    className="Timeline-Pointer-0-1-5"
    style={
      Object {
        "left": "0%",
      }
    }
  >
    <span
      className="Timeline-Timer-0-1-4"
    >
      00:00
    </span>
  </div>
  <div
    className="Highlight-Highlight-0-1-6 Highlight-HighlightRotated-0-1-7"
    style={
      Object {
        "left": "100%",
      }
    }
  >
    <button
      className="Highlight-Pointer-0-1-8"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      type="button"
    />
    <div
      className="Highlight-CaptionHolder-0-1-9"
    >
      <p>
        Lorem ipsum
      </p>
    </div>
  </div>
</div>
`;

exports[`Timeline tests should match snapshot without highlights 1`] = `
<div
  className="Timeline-Timeline-0-1-1"
>
  <div
    className="Timeline-Track-0-1-2"
  />
  <input
    className="Timeline-Track-0-1-2 Timeline-Range-0-1-3"
    defaultValue={0}
    max={100}
    min={0}
    onChange={[Function]}
    step={1}
    type="range"
  />
  <div
    className="Timeline-Pointer-0-1-5"
    style={
      Object {
        "left": "0%",
      }
    }
  >
    <span
      className="Timeline-Timer-0-1-4"
    >
      00:00
    </span>
  </div>
</div>
`;
